wayback-webapp/src/main/webapp/WEB-INF/query/BubbleCalendarHeatmap.jsp

```diff
@@ -0,0 +1,581 @@
+            }
+        }
+    }
+
+span = max - min + 1;
+
+// draw 12 months, 0-11 (0=Jan, 11=Dec)
+for(int moy = 0; moy < 12; moy++) {
+	Partition<Partition<CaptureSearchResult>> curMonth = data.monthsByDay.get(moy);
+	List<Partition<CaptureSearchResult>> monthDays = curMonth.list();
+%>
+    <div class="month" id="<%= data.yearNum %>-<%= moy %>">
+	    <table>
+	
+	       <thead>
+	           <tr>
+	               <th colspan="7"><span class="label"></span></th>
+	           </tr>
+	       </thead>
+	       <tbody>
+	           <tr>
+<%
+		cal.setTime(curMonth.getStart());
+		int skipDays = cal.get(Calendar.DAY_OF_WEEK) - 1;
+		int daysInMonth = cal.getActualMaximum(Calendar.DAY_OF_MONTH);
+		// skip until the 1st:
+		for(int i = 0; i < skipDays; i++) {
+			%><td><div class="date"></div></td><%
+		}
+		int dow = skipDays;
+		int dom;
+		for(dom = 0; dom < daysInMonth; dom++) {
+
+
+			int count = monthDays.get(dom).count();
+			if(count > 0) {
+				// one or more captures in this day:
+				CaptureSearchResult firstCaptureInDay = 
+					monthDays.get(dom).list().get(0);
+				String replayUrl = uriConverter.makeReplayURI(
+							firstCaptureInDay.getCaptureTimestamp(),
+							firstCaptureInDay.getOriginalUrl());
+				Date firstCaptureInDayDate = firstCaptureInDay.getCaptureDate();
+				String safeUrl = fmt.escapeHtml(replayUrl);
+                                double alpha = (double) (count - min + 1) / span;
+
+                                if(span >= 5)
+                                {
+                                    int step = span / 5;
+
+                                    if( count < (min + step) )
+                                        alpha = 0.2;
+                                    else if( count >= (min + step) && count < (min + (step*2)) )
+                                        alpha = 0.4;
+                                    else if(count >= (min + (step*2)) && count < (min + (step*3)))
+                                        alpha = 0.6;
+                                    else if(count >= (min + (step*3)) && count < (min + (step*4)))
+                                        alpha = 0.8;
+                                    else
+                                        alpha = 1;
+                                }
+				%><td>
+                            
+                            <div id="calendarBubble" style="background-color: rgba( 0, 171, 235, <%= alpha %> );"></div>
+                            
+				</td><%
+
+			} else {
+				// zero captures in this day:
+				%><td>
+	                <div class="date"></div>
+				</td><%
+				
+			}
+
+
+			if(((dom+skipDays+1) % 7) == 0) {
+				// end of the week, start a new tr:
+				%></tr><tr><%
+			}
+		}
+		// fill in blank days until the end of the current week:
+		while(((dom+skipDays) % 7) != 0) {
+			%><td></td><%
+			dom++;
+		}
+%>
+			   </tr>
+			</tbody>
+		</table>    
+      </div>
+    
+<%
+}
+
+min = 0;
+max = 0;
```

Comment:
Well, I just reset them to zero at 429-430 for next selected year. As minimum and maximum has been retrieved separately for each year.